common/config.js

```diff
@@ -75,6 +75,27 @@ function completeDirectorConfig(director) {
       cpi: info.cpi
     });
   }
+  let boshRateLimitEnabled = config.enable_bosh_rate_limit;
+  if (boshRateLimitEnabled) {
+    const maxWorkers = _.get(director, 'max_workers', 6);
+    const userCreateWorkers = _.get(director, 'policies.user.create');
+    const userUpdateWorkers = _.get(director, 'policies.user.update');
+    const userDeleteWorkers = _.get(director, 'policies.user.delete');
+    const autoWorkers = _.get(director, 'policies.scheduled.max_workers', (maxWorkers / 2));
+
+    if (userCreateWorkers === undefined || userUpdateWorkers === undefined || userDeleteWorkers === undefined) {
+      throw new Error('Invalid director config: user policy share numbers not defined');
+    }
+
+    if (_.sum([userCreateWorkers, userUpdateWorkers, userDeleteWorkers, autoWorkers]) > maxWorkers) {
+      throw new Error('Invalid director config: policy shares add up to more than max_workers count');
+    }
+
+    _.set(director, 'policies.user.create.max_workers', userCreateWorkers);
+    _.set(director, 'policies.user.update.max_workers', userUpdateWorkers);
+    _.set(director, 'policies.user.delete.max_workers', userDeleteWorkers);
+    _.set(director, 'policies.scheduled.max_workers', autoWorkers);
```

Comment:
why are these set required? you can as well retrieve the same as just `targetDirectorConfig.policies.user[action]`? seems dupe.